### PR TITLE
[apps] migrating requests logic to base class

### PR DESCRIPTION
**Pull request by ryandeivert**
to @austinbyers 
cc @airbnb/streamalert-maintainers 
size: small

### Changes
* To make the app class interface easier to subclass/utilize, I've moved the functionality
  for performing GET requests against an endpoint to within the base app class.
* Updating tests to account for the migrated code.
* Adding debug statement to log when a request occurs.

### CODE REVIEW

File: tests/unit/app_integrations/test_apps/test_app_base.py [hunk 1]
```diff
@@ -212,3 +212,18 @@ def test_gather_running(self, finalize_mock):
...
         self._app.gather()
         finalize_mock.assert_not_called()
+
+    @patch('requests.get')
+    def test_make_request_bad_response(self, requests_mock):
+        """App Integration - Make Request, Bad Response"""
+        requests_mock.return_value = Mock(
+            status_code=404,
+            json=Mock(side_effect=[{'message': 'something went wrong'}])
+        )
```
Comment (austinbyers) [inline tests/unit/app_integrations/test_apps/test_app_base.py]: Nice! I know this is copied, but I didn't know you could use a raw Mock like this

File: app_integrations/apps/duo.py [hunk 2]
```diff
@@ -116,13 +114,11 @@ def _get_duo_logs(self, hostname, full_url):
         if not headers:
             return False
 
-        # Perform the request and get the list of logs
-        response = requests.get(full_url, headers=headers, params=params)
-
-        if not self._check_http_response(response):
-            return False
+        # Make the request to the api, resulting in a bool or dict
+        response = self._make_request(full_url, headers=headers, params=params)
```
Comment (austinbyers) [inline app_integrations/apps/duo.py]: If `response` is False, you'll get a `KeyError` on the next line

Comment (ryandeivert) [reply app_integrations/apps/duo.py]: Nice good catch. Making that fix

**Comment by ryandeivert** [comment]
@austinbyers  added a commit and rebased

**Comment by mime-frame** [comment]
image

**Pull request merged**